static/app/components/carousel.tsx

```diff
@@ -0,0 +1,140 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [isAtStart, setIsAtStart] = useState<boolean>(true);
+  const [isAtEnd, setIsAtEnd] = useState<boolean>(false);
+  const [showArrows, setShowArrows] = useState<boolean>(true);
+
+  const setStartAndEnd = (scrollLeft: number, totalScrollDist: number) => {
+    // scrollLeft: the amount scrolled horizontally.
+    // start at 0 and end at totalScrollDist
+
+    // totalScrollDist = scrollWidth - clientWidth
+    // scrollWidth: total element width including overflow
+    // clientWidth: width of the element excluding overflow
+    if (scrollLeft < 10) {
```

Comment:
On second thoughts, let's remove the `10` since we're not accurate when scrolling around.